packages/insomnia-app/app/ui/components/editors/auth/o-auth-2-auth.js

```diff
@@ -483,6 +497,32 @@ class OAuth2Auth extends React.PureComponent<Props, State> {
     return { basic: basicFields, advanced: advancedFields };
   }
 
+  static renderExpiryFromJWT(token: string): React.Element<*> | string | null {
+    if (!token) {
+      return null;
+    }
+
+    const base64Url = token.split('.')[1];
```

Comment:
I just pulled this down locally to test and the Oauth2 tab fails to render for a brand new request as a result of this implementation. The following suggestion matches how `renderExpireAt` is written

```suggestion
  static renderExpiryFromJWT(token: ?OAuth2Token): React.Element<*> | string | null {
    if (!token || !token.identitiyToken) {
      return null;
    }

    const base64Url = token.identitiyToken.split('.')[1];
```

Reply:
I'll make this change. 
I was trying to keep the function generic, to be able to be used with any JWT :sweat_smile:
I guess since accessToken already has an expiry function, i'll make this one dedicated to identityToken.

Reply:
Fair enough 😅 it fails because likely somewhere in the render loop this logic is reached while token is `undefined`, and the previous code would fail with an error like `identityToken cannot be found on undefined`